### ISSUE TITLE
Support deploying external agents with the machines watcher

### ISSUE THREAD

**Issue by ripienaar** [choria-io/go-choria#2029]
Later we'll rename it to plugins watcher once its properly generic.

**Comment by jeffmccune** [comment]
You're probably aware, but just an FYI the `choria machine mms` command isn't plumbed to match the [docs](https://choria.io/docs/autoagents/watcher_reference/#autonomous-agent-archives) yet.  I took a stab but wasn't sure if this should be part of the new plugin system.  Only mention it in off chance it was overlooked.

**Comment by jeffmccune** [comment]
Never-mind, I see it: `choria machine plugins pack`

**Comment by ripienaar** [comment]
Yeah sorry about that been a bit of turbulence here as I realised it would be impossible to get the “machines” watcher to be more generic and keep working as before. 

So there is a plugins watcher now that’s almost identical but can also handle rpc agents and more. 

Docs not updated for that will try today. 

Such is life on main I am afraid
